ouroboros-network/src/Ouroboros/Network/Protocol/Trans/Hello/Util.hs

```diff
@@ -0,0 +1,159 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Utility module which provides wrappers which map a 'Peer' or 'PiplinedPeer'
+-- of an original protocol 'ps' into 'Peer' \/ 'PiplinedPeer' of @'Hello' ps
+-- stIdle@ protocol.
+--
+module Ouroboros.Network.Protocol.Trans.Hello.Util
+  ( -- * Wrap 'Peer' in 'Hello' protocol
+    wrapClientPeer
+  , wrapServerPeer
+
+    -- * Wrap 'PipelinedPeer' in 'Hello' protcol.
+  , wrapClientPeerPipelined
+  , wrapServerPeerPipelined
```

Comment:
Are these always going to be used qualified, or should we include "hello" in these names?

Reply:
Yes, I think this will lead to cleaner code base if we start using qualifications.